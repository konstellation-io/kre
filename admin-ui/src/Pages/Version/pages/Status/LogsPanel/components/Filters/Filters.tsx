import GroupSelect, {
  GroupSelectData
} from 'Components/Form/GroupSelect/GroupSelect';
import MultiSelect, {
  MultiSelectOption,
  SelectTheme
} from 'Components/Form/MultiSelect/MultiSelect';

import DateFilter from './components/DatesFilter/DateFilter';
import { GetLogTabs_logTabs_filters } from 'Graphql/client/queries/getLogs.graphql';
import Left from 'Components/Layout/Left/Left';
import LevelIcon from 'Components/LevelIcon/LevelIcon';
import { LogLevel } from 'Graphql/types/globalTypes';
import { NodeSelection } from 'Graphql/client/typeDefs';
import React from 'react';
import Right from 'Components/Layout/Right/Right';
import SearchSelect from 'Components/Form/SearchSelect/SearchSelect';
import cx from 'classnames';
import styles from './Filters.module.scss';
import useWorkflowsAndNodes from 'Hooks/useWorkflowsAndNodes';

// TODO: use GroupSelectData in the filters instead of [string]
function nodesSelectionToDoubleSelector(
  selections: NodeSelection[]
): GroupSelectData {
  const data: { [key: string]: string[] } = {};

  selections.forEach(({ workflowName, nodeNames }) => {
    data[workflowName] = nodeNames;
  });

  return data;
}

function doubleSelectorToNodesSelection(
  selections: GroupSelectData
): NodeSelection[] {
  const data: NodeSelection[] = Object.entries(selections).map(
    ([workflowName, nodeNames]) => ({
      workflowName,
      nodeNames,
      __typename: 'NodeSelection'
    })
  );

  return data;
}

const levelsOrdered = [
  LogLevel.INFO,
  LogLevel.DEBUG,
  LogLevel.WARN,
  LogLevel.ERROR
];

type Props = {
  versionId: string;
  updateFilters: Function;
  filterValues: GetLogTabs_logTabs_filters;
};
function Filters({ updateFilters, filterValues, versionId }: Props) {
  const { workflowsAndNodesNames } = useWorkflowsAndNodes(versionId);

  const levelOptions = levelsOrdered.map(
    (level: LogLevel) =>
      ({
        label: level,
        Icon: <LevelIcon level={level} />
      } as MultiSelectOption<LogLevel>)
  );

  function onNodeSelection(newSelection: GroupSelectData) {
    updateFilters({
      nodes: doubleSelectorToNodesSelection(newSelection)
    });
  }

  function onLevelSelection(newLevels: LogLevel[]) {
    updateFilters({ levels: newLevels });
  }

  function onSearchUpdate(newSearch: string) {
    updateFilters({ search: newSearch });
  }

  return (
    <div className={styles.container}>
      <Left className={cx(styles.filterContainer, styles.leftContainer)}>
        <div className={styles.searchFilter}>
          <SearchSelect
            options={[]}
            onChange={onSearchUpdate}
            value={filterValues.search}
            className={styles.searchForm}
            placeholder="Search"
            hideError
            hideLabel
            showSearchIcon
          />
        </div>
        <div className={styles.selectProcesses}>
          <GroupSelect
            options={workflowsAndNodesNames}
            formSelectedOptions={nodesSelectionToDoubleSelector(
              filterValues.nodes || []
            )}
            onChange={onNodeSelection}
            label=""
            placeholder="Select Processes"
            className={styles.selectProcessForm}
            hideError
            hideSelections
          />
        </div>
      </Left>
      <Right className={cx(styles.filterContainer, styles.rightFilters)}>
        <div className={styles.filterLevel}>
          <MultiSelect<LogLevel>
            onChange={onLevelSelection}
            label=""
            hideError
            placeholder="ALL LEVELS"
            selectionUnit="LEVEL"
            selectAllText="ALL LEVELS"
            options={levelOptions}
            formSelectedOptions={filterValues.levels || []}
            theme={SelectTheme.LIGHT}
          />
        </div>
        <DateFilter
          selectedOption={filterValues.dateOption}
          updateFilters={updateFilters}
          formStartDate={filterValues.startDate}
          formEndDate={filterValues.endDate}
        />
      </Right>
    </div>
  );
}

export default Filters;
