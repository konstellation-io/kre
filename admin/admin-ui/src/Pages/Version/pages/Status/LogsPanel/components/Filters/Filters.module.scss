@import 'Styles/colors';
@import 'Styles/mixins';
@import 'Styles/variables';

$input-height: 30px;

.container {
  @include font-body;
  line-height: 0;

  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: space-between;
  height: auto;
  padding: 0 20px;
  flex-shrink: 0;
}

.filterContainer {
  display: flex;
  flex-direction: row;
  align-items: center;

  > div {
    margin-right: $grid-unit;
  }
}
.rightFilters {
  flex-shrink: 0;

  .filterLevel {
    width: 142px;
  }

  .dateFilter {
    width: $grid-unit * 21.5;
  }
}

.leftContainer {
  flex-grow: 1;
  flex-basis: 0;
  max-width: 100 * $grid-unit;

  .searchFilter {
    width: 60%;
    height: $input-height;
    min-width: 100px;
    margin: $grid-unit $grid-unit $grid-unit 0;

    .searchForm {
      input {
        height: $input-height;
        min-height: $input-height;
      }
    }
  }

  .selectProcesses {
    width: 40%;
    height: $input-height;
    min-width: 170px;

    .selectProcessForm > div:first-child {
      height: $input-height;
      min-height: $input-height;
      line-height: $input-height - 1px;
    }
  }
}
