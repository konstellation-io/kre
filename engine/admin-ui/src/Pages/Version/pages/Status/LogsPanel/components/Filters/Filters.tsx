import {
  GroupSelect,
  GroupSelectData,
  Left,
  MultiSelect,
  MultiSelectOption,
  MultiSelectTheme,
  Right,
  SearchSelect
} from 'kwc';

import DateFilter from './components/DatesFilter/DateFilter';
import LevelIcon from 'Components/LevelIcon/LevelIcon';
import { LogLevel } from 'Graphql/types/globalTypes';
import { LogPanelFilters, NodeSelection } from 'Graphql/client/typeDefs';
import React from 'react';
import cx from 'classnames';
import styles from './Filters.module.scss';
import useWorkflowsAndNodes from 'Hooks/useWorkflowsAndNodes';

// TODO: use GroupSelectData in the filters instead of [string]
function nodesSelectionToDoubleSelector(
  selections: NodeSelection[]
): GroupSelectData {
  const data: { [key: string]: string[] } = {};

  selections.forEach(({ workflowName, nodeNames }) => {
    data[workflowName] = nodeNames;
  });

  return data;
}

function doubleSelectorToNodesSelection(
  selections: GroupSelectData
): NodeSelection[] {
  const data: NodeSelection[] = Object.entries(selections).map(
    ([workflowName, nodeNames]) => ({
      workflowName,
      nodeNames,
      __typename: 'NodeSelection'
    })
  );

  return data;
}

const levelsOrdered = [
  LogLevel.INFO,
  LogLevel.DEBUG,
  LogLevel.WARN,
  LogLevel.ERROR
];

type Props = {
  versionName: string;
  updateFilters: Function;
  filterValues: LogPanelFilters;
  runtimeId: string;
};
function Filters({ updateFilters, filterValues, versionName, runtimeId }: Props) {
  const { workflowsAndNodesNames } = useWorkflowsAndNodes(versionName, runtimeId);

  const levelOptions = levelsOrdered.map(
    (level: LogLevel) =>
      ({
        label: level,
        Icon: <LevelIcon level={level} />
      } as MultiSelectOption<LogLevel>)
  );

  function onNodeSelection(newSelection: GroupSelectData) {
    updateFilters({
      nodes: doubleSelectorToNodesSelection(newSelection)
    });
  }

  function onLevelSelection(newLevels: LogLevel[]) {
    updateFilters({ levels: newLevels });
  }

  function onSearchUpdate(newSearch: string) {
    updateFilters({ search: newSearch });
  }

  return (
    <div className={styles.container}>
      <Left className={cx(styles.filterContainer, styles.leftContainer)}>
        <div className={styles.searchFilter}>
          <SearchSelect
            options={[]}
            onChange={onSearchUpdate}
            value={filterValues.search}
            className={styles.searchForm}
            placeholder="Search"
            hideError
            hideLabel
            showSearchIcon
          />
        </div>
        <div className={styles.selectProcesses}>
          <GroupSelect
            options={workflowsAndNodesNames}
            formSelectedOptions={nodesSelectionToDoubleSelector(
              filterValues.nodes || []
            )}
            onChange={onNodeSelection}
            label=""
            placeholder="Select Processes"
            className={styles.selectProcessForm}
            hideError
            hideSelections
          />
        </div>
      </Left>
      <Right className={cx(styles.filterContainer, styles.rightFilters)}>
        <div className={styles.filterLevel}>
          <MultiSelect<LogLevel>
            onChange={onLevelSelection}
            label=""
            hideError
            placeholder="ALL LEVELS"
            selectionUnit="LEVEL"
            selectAllText="ALL LEVELS"
            options={levelOptions}
            formSelectedOptions={filterValues.levels || []}
            theme={MultiSelectTheme.LIGHT}
          />
        </div>
        <DateFilter
          selectedOption={filterValues.dateOption}
          updateFilters={updateFilters}
          formStartDate={filterValues.startDate}
          formEndDate={filterValues.endDate}
        />
      </Right>
    </div>
  );
}

export default Filters;
