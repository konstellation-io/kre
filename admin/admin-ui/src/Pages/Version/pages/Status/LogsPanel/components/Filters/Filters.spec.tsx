import React from 'react';
import Filters from './Filters';
import { shallow } from 'enzyme';

jest.mock('@apollo/react-hooks', () => ({
  useQuery: jest.fn(() => ({}))
}));

describe('Filters', () => {
  let wrapper;

  const onDateChangeMock = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Filters
        onDateChange={onDateChangeMock}
        filterValues={{
          startDate: '2020-01-01',
          endDate: '2020-01-01',
          dateOption: '2020-01-01'
        }}
      />
    );
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
