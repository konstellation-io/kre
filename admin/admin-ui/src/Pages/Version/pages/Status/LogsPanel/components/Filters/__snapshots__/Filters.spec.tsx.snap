// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters matches snapshot 1`] = `
<div
  className="container"
>
  <Left
    className="filterContainer leftContainer"
  >
    <div
      className="searchFilter"
    >
      <SearchSelect
        className="searchForm"
        hideError={true}
        hideLabel={true}
        onChange={[Function]}
        options={Array []}
        placeholder="Search"
        showSearchIcon={true}
      />
    </div>
    <div
      className="selectProcesses"
    >
      <GroupSelect
        className="selectProcessForm"
        formSelectedOptions={Object {}}
        hideError={true}
        hideSelections={true}
        label=""
        onChange={[Function]}
        options={
          Object {
            "": Array [
              "Entry points",
            ],
          }
        }
        placeholder="Select Processes"
      />
    </div>
  </Left>
  <Right
    className="filterContainer rightFilters"
  >
    <div
      className="filterLevel"
    >
      <MultiSelect
        formSelectedOptions={Array []}
        hideError={true}
        label=""
        onChange={[Function]}
        options={
          Array [
            Object {
              "Icon": <LevelIcon
                level="INFO"
              />,
              "label": "INFO",
            },
            Object {
              "Icon": <LevelIcon
                level="DEBUG"
              />,
              "label": "DEBUG",
            },
            Object {
              "Icon": <LevelIcon
                level="WARN"
              />,
              "label": "WARN",
            },
            Object {
              "Icon": <LevelIcon
                level="ERROR"
              />,
              "label": "ERROR",
            },
          ]
        }
        placeholder="ALL LEVELS"
        selectAllText="ALL LEVELS"
        selectionUnit="LEVEL"
        theme="light"
      />
    </div>
    <DateFilter
      formEndDate="2020-01-01"
      formStartDate="2020-01-01"
      selectedOption="2020-01-01"
    />
  </Right>
</div>
`;
